client/src/modules/util/pagination.tsx

```diff
@@ -0,0 +1,45 @@
+import { Button } from '@chakra-ui/react';
+import React from 'react';
+
+interface PaginationProps<T, V> {
+  data: Array<T>;
+  dataFlattener: (page: T) => Array<V>;
+  displayOnEmpty?: React.ReactNode;
+  itemsPerPage: number;
+  mapper: (item: V) => React.ReactNode;
+  onClickForMore: (offset: number) => void;
+  totalItemsFromData: (data: Array<T>) => number;
+}
+
+export const Pagination = <T, V>({
+  data,
+  dataFlattener,
+  displayOnEmpty,
+  itemsPerPage,
+  mapper,
+  onClickForMore,
+  totalItemsFromData,
+}: PaginationProps<T, V>) => {
+  const flattenedData = data.flatMap(dataFlattener);
+  const currentPage = Math.ceil(flattenedData.length / itemsPerPage);
+  const totalPages = Math.ceil(totalItemsFromData(data) / itemsPerPage);
```

Comment:
This approach seems a bit abstract. All the pagination component really needs is an array of things to render, a total, `itemsPerPage`, `onClickForMore` and `displayOnEmpty`. `data`, `dataFlattener` and `mapper` could just be a single `items` array, that's the result of

```
data.flatMap(dataFlattener).map(mapper)
```

That way, the Pagination component's really focused on pagination specific stuff, not on what's being paginated.

Reply:
is the current Pagination good enough? 🤔

Reply:
It is now :) I figured you'd been through enough, so I just pushed the nitpicks.